Tests/MongoSwiftTests/TestSetup.swift

```diff
@@ -0,0 +1,13 @@
+import Foundation
+import MongoSwift
+
+/// This should be assigned to NSPrincipalClass in the project's plist file
+class TestSetup: NSObject {
```

Comment:
@kmahar: I believe we'll want this class referenced from https://github.com/mongodb/mongo-swift-driver/commit/0bb9776fba22f9e5abbec43c4f23bf61b0a6be9c#diff-47062118b9b95971fa9ba0ca24e2270eR22 (or whatever is used to generate that).

For context, see the final section of https://stackoverflow.com/a/41895011/162228.

Reply:
So this seems reasonable for ensuring things happen correctly when running the tests in Xcode. And we should be able to edit the .plist files using the ruby gem I mentioned earlier. 

However, now that I think about it, I am not sure that the .plist file is actually considered at all when you run the tests from the command line, as you can run tests there without having ever generated the .xcodeproj. 

So we may need an alternate approach in that case...

Reply:
Does this mean that there are effectively three variations for running tests?

 1. Xcode in Darwin, which uses the plist file.
 2. `swift test` in Darwin, which automatically collects tests.
 3. `swift test` in Linux, which requires `LinuxMain.swift` to manually collect tests.

While this approach theoretically ensures that cleanup happens via `deinit`, I'll note my other argument that cleanup really isn't essential since any "leaked" memory will be freed when the process exits. If we go the route of simply calling `MongoSwift.initialize()` from `MongoClient` (as you suggested [here](https://github.com/mongodb/mongo-swift-driver/pull/73#discussion_r194542780)) then we can likely avoid any test setup for all three variants.

Reply:
yes, those are the three. given all these testing variations, I think I like the calling `MongoSwift.initialize()` from `MongoClient` approach best, let's see what @mbroadst thinks.

Reply:
@mbroadst: To clarify, we're not suggesting calling `mongoc_init()` directly from `MongoClient.init` and relying purely on libmongoc to ensure at-most-once behavior. I'd still plan to use the Swift singleton and leave the original `MongoSwift.initialize()` and `MongoSwift.cleanup()` functions in place; however, `MongoClient.init` could call `MongoSwift.initialize()` each time in case a user forgets (or doesn't want to) call it themselves. I think we could still suggest that user's manually call both methods at the start/end of their application, respectively.

Reply:
After much contemplation I think the only way forward is to make initialization explicit. It's very tempting to bury the initialization call in the initializer for `MongoClient`, however it really ties our hands for future versions of the driver and language (perhaps Swift will support deinitializing singletons in the future, for instance). The only way to provide a consistent experience here is to provide explicit initialize/cleanup methods, and document them appropriately.